pkg/gitops/audit_handler.go

```diff
@@ -14,6 +14,74 @@
 
 package gitops
 
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/klog/v2"
+
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+)
+
 func (cr *CustomRepo) HandleEventList(jsonstring []byte) error {
+	cr.Mutex.Lock()
+	defer cr.Mutex.Unlock()
+	eventList := auditv1.EventList{}
+	jsonstring = bytes.TrimPrefix(jsonstring, []byte("\xef\xbb\xbf"))
+	err := json.Unmarshal(jsonstring, &eventList)
+	if err != nil {
+		klog.ErrorS(err, "unable to unmarshal json into event list struct")
+		return err
+	}
+	for _, event := range eventList.Items {
+		if event.Stage != "ResponseComplete" ||
+			event.ResponseStatus.Status == "Failure" ||
+			event.User.Username == cr.ServiceAccount {
+			klog.V(4).InfoS("Audit event skipped (audit Stage != ResponseComplete, audit ResponseStatus != Success, or audit produced by rollback)")
+			continue
+		}
+		if cr.RollbackMode {
+			klog.V(2).Infof("Rollback currently in progress, rejecting audit")
```

Comment:
I left in a bunch of klog.Infof()'s and klog.Errorf()'s in my portions of the code that we should change to klog.InfoS()'s and klog.ErrorS()'s

Reply:
Ok, will clean this up.